dali/imgcodec/util/convert.h

```diff
@@ -75,16 +129,95 @@ inline void ConvertDType(Out *out, const In *in) {
   *out = ConvertSatNorm<Out>(*in);
 }
 
+/**
+ * @brief Returns a color space conversion function to use with Convert.
+ */
+template <typename Out, typename In>
+inline std::function<void(Out *, const In *)> GetConversionFunc(
+    DALIImageType out_format, ptrdiff_t out_channel_stride,
+    DALIImageType in_format, ptrdiff_t in_channel_stride) {
```

Comment:
Sorry, but this means that the per-pixel processing now goes through a dynamically allocated, type erased thing (std::function) that uses function pointers. The performance will be unacceptable.

Reply:
You're absolutely right. I redesigned the code to eliminate this